### PR TITLE
OPC user edit throwing hydrate errors fixes

### LINKED ISSUE #53: chef-server-ctl user-edit throwing hydrate errors
Details in https://github.com/chef/chef-server/issues/1890

Can continue discussion here, since this seems to be a knife-opc issue.

### PR DESCRIPTION
**Pull request by antima-gupta**
Signed-off-by: antima-gupta <agupta@msystechnologies.com>

### Description
- Fixed `knife opc user edit USERNAME`
- Fixed `knife opc user edit USERNAME -f FILENAME`

### Issues Resolved
#53
https://github.com/chef/chef-workstation/issues/1167 
https://github.com/chef/chef-workstation/issues/1179#issuecomment-629321414

### Check List

- [ ] New functionality includes tests
- [x] All tests pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [x] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by dheerajd-msys** [comment]
@btm @PrajaktaPurohit Can someone review this PR?

File: README.md [hunk 1]
```diff
@@ -74,7 +74,10 @@ Creates a new user in your OPC installation. The user's private key will be retu
 
 Deletes the given OPC user.
 
-### knife opc user edit USERNAME
+### knife opc user edit USERNAME (options)
+
+- `-i FILENAME`, `--input FILENAME`: Will read the user information (i.e. username, email, display_name, first_name, last_name, middle_name) from the input FILENAME and update the given OPC user.
```
Comment (kagarmoe) [inline README.md:79]:
```suggestion
- `-i FILENAME`, `--input FILENAME`: Reads the user information (`username`, `email`, `display_name`, `first_name`, `last_name`, `middle_name`) from the input FILENAME and updates the given OPC user.
```

File: README.md [hunk 2]
```diff
@@ -74,7 +74,10 @@ Creates a new user in your OPC installation. The user's private key will be retu
...
+- `-f FILENAME`, `--filename FILENAME`: Open FILENAME on $EDITOR. When finished, Updates the given OPC user.
```
Comment (kagarmoe) [inline README.md:80]:
```suggestion
- `-f FILENAME`, `--filename FILENAME`: Opens FILENAME in $EDITOR. Updates the given OPC user when the file is saved.
```

Comment (kagarmoe) [reply README.md:80]: Is the update on save or close?

Comment (antima-gupta) [reply README.md:80]: @kagarmoe Update is on save and close the $Editor.

File: README.md (hunk 1 above)
Comment (antima-gupta) [reply README.md]: Agreed,I  updated it.

File: README.md [hunk 3]
```diff
@@ -82,7 +82,10 @@ Creates a new user in your OPC installation. The user's private key will be retu
 
 Deletes the given OPC user.
 
-### knife opc user edit USERNAME
+### knife opc user edit USERNAME (options)
+
+- `-i FILENAME`, `--input FILENAME`: Reads the user information (`username`, `email`, `display_name`, `first_name`, `last_name`, `middle_name`) from the input FILENAME and updates the given OPC user.
```
Comment (PrajaktaPurohit) [inline README.md:87]: It would be helpful to mention that this file is expected to be in the json format and optionally add an example.

Comment (antima-gupta) [reply README.md:87]: Agreed, I updated it.

File: README.md [hunk 4]
```diff
@@ -82,7 +82,25 @@ Creates a new user in your OPC installation. The user's private key will be retu
 
 Deletes the given OPC user.
 
-### knife opc user edit USERNAME
+### knife opc user edit USERNAME (options)
+
+- `-i FILENAME`, `--input FILENAME`: Reads the user information (`username`, `email`, `display_name`, `first_name`, `last_name`, `middle_name`) from the input FILENAME and updates the given OPC user. This file is expected to be in the json format.
```
Comment (IanMadd) [inline README.md:87]:
```suggestion
- `-i FILENAME`, `--input FILENAME`: Reads the user information (`username`, `email`, `display_name`, `first_name`, `last_name`, `middle_name`) from the input FILENAME and updates the given OPC user. Knife will accept a JSON file.
```

Comment (antima-gupta) [reply README.md]: @IanMadd I updated the changes.Please have a look.

**Pull request merged**